### PR TITLE
util/wrapper: Add Static Client wrapper

### PR DESCRIPTION
**Pull request by ben-toogood**
A replacement for the selector/static which doesn't involve overriding the service name (this improves tracing among other things).

Example usage:
`client := pb.NewAuthService("go.micro.auth", wrapper.StaticClient("auth-srv:8010", options.Client))`

The selector/selector package is soon to be deprecated, so those who were using this previously should no longer use MICRO_SELECTOR=static, and instead wrap their clients as demonstrated above. In go-micro v2.9 and above, this will be done automatically when the MICRO_SELECTOR=static flag is detected (a depreciation warning will be logged), however this will be removed in v3.

### CODE REVIEW

**Comment by vtolstov** [comment]
tests added

File: server/grpc/server.go [hunk 1]
```diff
@@ -97,7 +97,7 @@ func prepareEndpoint(method reflect.Method) *methodType {
 		streamType := reflect.TypeOf((*server.Stream)(nil)).Elem()
 		if !argType.Implements(streamType) {
 			if logger.V(logger.ErrorLevel, logger.DefaultLogger) {
-				logger.Error(mname, "argument does not implement Streamer interface:", argType)
+				logger.Error(mname, " argument does not implement Streamer interface: ", argType)
```
Comment (domwong) [inline server/grpc/server.go]: Let's switch this to `logger.Errorf`

Comment (domwong) [reply server/grpc/server.go]: In fact can we change all of the new logger invocations in this file please

File: server/rpc_router.go [hunk 2]
```diff
@@ -148,38 +148,38 @@ func prepareMethod(method reflect.Method) *methodType {
 		// check stream type
 		streamType := reflect.TypeOf((*Stream)(nil)).Elem()
 		if !argType.Implements(streamType) {
-			log.Error(mname, "argument does not implement Stream interface:", argType)
+			log.Error(mname, " argument does not implement Stream interface: ", argType)
```
Comment (domwong) [inline server/rpc_router.go]: Ditto these logs

**Review by domwong — APPROVED** [review]
Looks fine. Just change those log lines to use the formatter versions

**Pull request merged**